src/payment/apple-pay/ApplePayButton.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { performApplePayPayment } from './service';
+
+
+export default class ApplePayButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      canMakePayments: global.ApplePaySession && global.ApplePaySession.canMakePayments(),
+    };
+  }
+
+  handleClick = () => {
+    performApplePayPayment({
+      totalAmount: this.props.totalAmount,
+      onPaymentBegin: this.props.onPaymentBegin,
+      onPaymentComplete: this.props.onPaymentComplete,
+      onMerchantValidationFailure: this.props.onMerchantValidationFailure,
+      onPaymentAuthorizationFailure: this.props.onPaymentAuthorizationFailure,
+      onPaymentCancel: this.props.onPaymentCancel,
+    });
+  }
+
+  render() {
+    if (!this.state.canMakePayments || this.props.totalAmount === undefined) return null;
+
+    const other = { ...this.props };
+
+    delete other.totalAmount;
+    delete other.onPaymentBegin;
+    delete other.onPaymentComplete;
+    delete other.onMerchantValidationFailure;
+    delete other.onPaymentAuthorizationFailure;
+    delete other.onPaymentCancel;
```

Comment:
What's going on here with these `delete`s?

Reply:
I want users to pass any html attribute through if they want to, but these props are not valid. I could use lodash's .omit() instead but opted not to use a library